src/Parser/TypeParser.php

```diff
@@ -139,26 +139,29 @@ private function parseAtomic(TokenIterator $tokens): Ast\Type\TypeNode
 			$tokens->dropSavePoint(); // because of ConstFetchNode
 		}
 
-		$exception = new ParserException(
-			$tokens->currentTokenValue(),
-			$tokens->currentTokenType(),
-			$tokens->currentTokenOffset(),
-			Lexer::TOKEN_IDENTIFIER
-		);
+		$tokensCopy = $tokens->copy();
```

Comment:
Can you not just `clone $tokens`?

Reply:
Yes, but I know Ondřej [dislikes `clone`](https://github.com/phpstan/phpstan-src/pull/1159#discussion_r841027012)